test/web/Program.cs

```diff
@@ -34,38 +34,45 @@ public static IHostBuilder CreateHostBuilder(string[] args, CommandLineOptions c
                 .ConfigureWebHostDefaults(webBuilder =>
                 {
                     webBuilder.UseStartup<Startup>();
-                    if (commandLineOptions.SocketEngine == SocketEngineType.IOUringTransport)
+
+                    switch (commandLineOptions.SocketEngine)
                     {
-                        webBuilder.ConfigureServices(serviceCollection =>
-                            serviceCollection.AddIoUringTransport(options =>
+                        case SocketEngineType.IOUringTransport:
+                            webBuilder.ConfigureServices(serviceCollection =>
+                                serviceCollection.AddIoUringTransport(options =>
+                                {
+                                    options.ThreadCount = commandLineOptions.ThreadCount;
+                                    options.ApplicationSchedulingMode = commandLineOptions.ApplicationCodeIsNonBlocking.Value ?
+                                        PipeScheduler.Inline : PipeScheduler.ThreadPool;
+                                }));
+                            break;
+                        case SocketEngineType.LinuxTransport:
+                            webBuilder.UseLinuxTransport(options =>
                             {
                                 options.ThreadCount = commandLineOptions.ThreadCount;
-                                options.ApplicationSchedulingMode = commandLineOptions.ApplicationCodeIsNonBlocking.Value ?
-                                    PipeScheduler.Inline : PipeScheduler.ThreadPool;
-                            }));
-                    }
-                    else if (commandLineOptions.SocketEngine == SocketEngineType.LinuxTransport)
-                    {
-                        webBuilder.UseLinuxTransport(options =>
-                        {
-                            options.ThreadCount = commandLineOptions.ThreadCount;
-                            options.DeferSend = commandLineOptions.DeferSends.Value;
-                            options.ApplicationSchedulingMode= commandLineOptions.ApplicationCodeIsNonBlocking.Value ?
+                                options.DeferSend = commandLineOptions.DeferSends.Value;
+                                options.ApplicationSchedulingMode= commandLineOptions.ApplicationCodeIsNonBlocking.Value ?
                                     PipeScheduler.Inline : PipeScheduler.ThreadPool;
-                        });
-                    }
-                    else
-                    {
-                        webBuilder.UseLinuxAsyncSockets(options =>
+                            });
+                            break;
+                        case SocketEngineType.DefaultSockets:
+                            webBuilder.UseSockets(options =>
                             {
-                                options.DispatchContinuations = commandLineOptions.DispatchContinuations.Value;
```

Comment:
@antonfirsov fyi, we've added an option for `DontAllocateMemoryForIdleConnections` in https://github.com/dotnet/aspnetcore/pull/19396.